src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleaseFileController.cs

```diff
@@ -152,10 +153,10 @@ public async Task<ActionResult<IEnumerable<FileInfo>>> GetAncillaryFiles(Guid re
         public async Task<ActionResult<FileInfo>> ReplaceAncillary(
```

Comment:
Rather than make this a 'replacement', would suggest we change this to be an 'update' across the controller and service

Reply:
I did this, but didn't like the result. I've pushed what I did to the `EES-4393-2` branch.

I think because the existing update is also being called to update a release data file's title, so to have a single `ReleaseFileService#Update`, meant extra complexity, and the code was quite hard to understand, given that it's being used in different ways across the service. I believe we've had dangerous bugs in the past where someone changes a method not realising it's being called somewhere else, and it breaks a part of the service we wouldn't expect.

So I think I'll start again, and have `ReleaseFileService` have separate `UpdateDataFile` and `UpdateAncillaryFile` methods, and validate that those methods are only being used with files of those types.

Reply:
I talked this through with Ben, and ended up restoring the status quo and going for a dead simple fix.

The work I did is still on the `EES-4393-2` branch - happy to move ahead with that or something else if you'd prefer.

Reply:
Discussed this with Nick on a call. I'll restore my initial changes when creating this PR, and then make the following changes:
- Make a single endpoint to update ancillary files that can both update details and replace an ancillary file.
- Rename the replace ancillary methods to update ancillary.
- Don't create a new release file - still create a new file, but just update the existing release file.
- Change the existing update endpoint to only update data file details, as it will only be used to update data files name/summary after these changes.